src/compiler/checker.ts

```diff
@@ -29768,7 +29817,7 @@ namespace ts {
             }
 
             const thisType = getThisTypeOfSignature(signature);
-            if (thisType) {
+            if (thisType && couldContainTypeVariables(thisType)) {

```

Comment:
Are these new guards on `couldContainTypeVariables` necessary for this to work, or are they an optimization?

Reply:
Purely an optimization.